### PR TITLE
added gitignore to repo

### PR DESCRIPTION
**Pull request by hanneshapke**
## Description

Seems like the repo was missing a gitignore file since it was a copy from the internal git repo at Google.

related https://github.com/tensorflow/model-card-toolkit/discussions/241

## Checklist

Before submitting a pull request, please be sure to do the following:
- [x] Read the [How to Contribute guide](https://github.com/tensorflow/model-card-toolkit/blob/master/CONTRIBUTING.md)
      if this is your first contribution.
- [x] Open an issue to discuss this change.
- [x] Update documentation if applicable.
- [x] Write new tests if applicable.

### CODE REVIEW

**Comment by codesue** [comment]
Thanks for adding this! I'd prefer using [GitHub's Python gitignore](https://github.com/github/gitignore/blob/main/Python.gitignore) as the base since it's more commonly used and doesn't require attribution. From there, we can add:

```
# VSCode project settings
.vscode

# Bazel generated files
bazel-*
**/*_pb2.py
**/*_pb2_grpc.py
```

And change
```
*.py[cod]
*$py.class

# C extensions
*.so
```
to
```
**/*.py[cod]
**/*$py.class

# C extensions
**/*.so
```

This is similar to [tfx/.gitignore](https://github.com/tensorflow/tfx/blob/master/.gitignore) but with more recent changes to GitHub's Python base incorporated.

**Comment by hanneshapke** [comment]
Thank you @codesue for your feedback. I updated the gitignore file.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+# VSCode project settings
+.vscode
+
+# Bazel generated files
+bazel-*
+**/*_pb2.py
+**/*_pb2_grpc.py
+
+# OS generated files
+**/.DS_Store
```
Comment (codesue) [inline .gitignore:164]: Nice! I forgot about this.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+# .python-version
```
Comment (codesue) [inline .gitignore:88]: Let's uncomment `.python-version`.

**Review by codesue — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (hanneshapke) [reply .gitignore]: Done :)

**Pull request merged**